sdk/exportentities/v1/workflow.go

```diff
@@ -0,0 +1,540 @@
+				HookModelName: h.Model,
+			}
+
+			if h.Conditions != nil {
+				hook.Conditions = *h.Conditions
+			}
+			n.Hooks = append(n.Hooks, hook)
+		}
+	}
+
+	if len(w.PipelineHooks) > 0 {
+		//Only one node workflow
+		addHooks(w.PipelineHooks)
+		return
+	}
+
+	addHooks(w.Hooks[n.Name])
+}
+
+func (e *NodeEntry) processNode(name string, w *sdk.Workflow) (bool, error) {
+	// Find WorkflowNodeAncestors
+	exist, err := e.processNodeAncestors(name, w)
+	if err != nil {
+		return false, err
+	}
+
+	if exist {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (e *NodeEntry) processNodeAncestors(name string, w *sdk.Workflow) (bool, error) {
+	var ancestorsExist = true
+	var ancestors []*sdk.Node
+
+	if len(e.DependsOn) == 1 {
+		a := e.DependsOn[0]
+		//Looking for the ancestor
+		ancestor := w.WorkflowData.NodeByName(a)
+		if ancestor == nil {
+			ancestorsExist = false
+		}
+		ancestors = append(ancestors, ancestor)
+	} else {
+		for _, a := range e.DependsOn {
+			//Looking for the ancestor
+			ancestor := w.WorkflowData.NodeByName(a)
+			if ancestor == nil {
+				ancestorsExist = false
+				break
+			}
+			ancestors = append(ancestors, ancestor)
+		}
+	}
+
+	if !ancestorsExist {
+		return false, nil
+	}
+
+	n, err := e.getNode(name)
+	if err != nil {
+		return false, err
+	}
+
+	switch len(ancestors) {
+	case 0:
+		w.WorkflowData.Node = *n
+		return true, nil
+	case 1:
+		w.AddTrigger(ancestors[0].Name, *n)
+		return true, nil
+	default:
+		if n != nil && n.Type == sdk.NodeTypeJoin && joinAsNode(n) {
+			w.WorkflowData.Joins = append(w.WorkflowData.Joins, *n)
+			return true, nil
+		}
+	}
+
+	// Compute join
+
+	// Try to find an existing join with the same references
+	var join *sdk.Node
+	for i := range w.WorkflowData.Joins {
+		j := &w.WorkflowData.Joins[i]
+		var joinFound = true
+
+		for _, ref := range j.JoinContext {
+			var refFound bool
+			for _, a := range e.DependsOn {
+				if ref.ParentName == a {
+					refFound = true
+					break
+				}
+			}
+			if !refFound {
+				joinFound = false
+				break
+			}
+		}
+
+		if joinFound {
+			j.Ref = fmt.Sprintf("fakeRef%d", e.ID)
+			join = j
+		}
+	}
+
+	var appendJoin bool
+	if join == nil {
+		joinContext := make([]sdk.NodeJoin, 0, len(e.DependsOn))
+		for _, d := range e.DependsOn {
+			joinContext = append(joinContext, sdk.NodeJoin{
+				ParentName: d,
+			})
+		}
+		join = &sdk.Node{
+			JoinContext: joinContext,
+			Type:        sdk.NodeTypeJoin,
+			Ref:         fmt.Sprintf("fakeRef%d", e.ID),
+		}
+		appendJoin = true
+	}
+
+	join.Triggers = append(join.Triggers, sdk.NodeTrigger{
+		ChildNode: *n,
+	})
+
+	if appendJoin {
+		w.WorkflowData.Joins = append(w.WorkflowData.Joins, *join)
+	}
+	return true, nil
+}
+
+func joinAsNode(n *sdk.Node) bool {
+	return n.Context != nil && (n.Context.Conditions.LuaScript != "" || len(n.Context.Conditions.PlainConditions) > 0)
+}
+
+func (w Workflow) GetName() string {
+	return w.Name
+}
+
+func (w Workflow) GetVersion() string {
```

Comment:
idem

Reply:
we must know the version to use the good struct to unmarshal